puppet/modules/web/templates/deploy-rpmrepo.sh.epp

```diff
@@ -0,0 +1,123 @@
+<%- |
+  Stdlib::Fqdn $servername,
+  Stdlib::Absolutepath $rpm_directory,
+  Stdlib::Absolutepath $rpm_source_directory,
+| -%>
+#!/bin/bash
+
+set -xe
+# This is a forced SSH command - uncomment to test locally
+set -f -- $SSH_ORIGINAL_COMMAND
+
+prepcache() {
+	if [[ -e $REPO_PATH ]]; then
+		if [[ $MERGE == false ]] && [[ $OVERWRITE == false ]] ; then
+			echo "Repo overwrite (${OVERWRITE}) and merge (${MERGE}) are disabled, but ${REPO_PATH} already exists"
+			exit 1
+		fi
+		cp -al $REPO_PATH "$REPO_INSTANCE_PATH"
+	else
+		mkdir -p $REPO_INSTANCE_PATH
+	fi
+}
+
+do_rsync() {
+	opts=(--archive --verbose --hard-links --log-file "$REPO_RSYNC_LOG")
+	if [[ $MERGE != true ]] ; then
+		opts+=('--delete')
+	fi
+
+	for ARCH in x86_64 source; do
+		rsync "${opts[@]}" --log-file-format "CHANGED ${ARCH}/%n" "${REPO_SOURCE_RPM}/${ARCH}/" "${REPO_INSTANCE_PATH}/${ARCH}/"
+	done
+
+	set +f
+	for d in "${REPO_INSTANCE_PATH}"/*; do
+		(
+			cd "$d"
+
+			latest=$(ls -t foreman-release-[0-9]*.rpm 2>/dev/null | head -n1)
+			if [[ -n "$latest" ]] ; then
+				ln -sf "$latest" foreman-release.rpm
+			fi
+
+			latest=$(ls -t foreman-client-release-[0-9]*.rpm 2>/dev/null | head -n1)
+			if [[ -n "$latest" ]] ; then
+				ln -sf "$latest" foreman-client-release.rpm
+			fi
+
+			latest=$(ls -t katello-repos-[0-9]*.rpm 2>/dev/null | head -n1)
+			if [[ -n "$latest" ]] ; then
+				ln -sf "$latest" katello-repos-latest.rpm
```

Comment:
While we're making changes, should we align `katello-repos` to `foreman-release` and `foreman-client-release` by dropping the `-latest` suffix?
```suggestion
				ln -sf "$latest" katello-repos.rpm
```

Alternatively, I wouldn't mind introducing `foreman-katello-release` as a subpackage of `foreman-release`. Possibly include all repos it needs in there and make it conflict with `foreman-release`.